framework/helpers/BaseHtml.php

```diff
@@ -1719,7 +1719,8 @@ public static function renderSelectOptions($selection, $items, &$tagOptions = []
         );
         $options['encode'] = ArrayHelper::getValue($options, 'encode', $encode);
 
-        if (null !== ($prompt = ArrayHelper::remove($tagOptions, 'prompt'))) {
+        $prompt = ArrayHelper::remove($tagOptions, 'prompt');
+        if (null !== $prompt) {
```

Comment:
The readability part is debatable. Specially when you deal with many `if` statements.

```php
if (nulll === $value) {
   // ...
}
if (0 === $value) {
   // ...
}
if (1 === $value) {
   // ...
}
if ('' === $value) {
   // ...
}
```